crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -39,7 +39,89 @@ use syn::{parse_macro_input, DeriveInput};
+///   the type's [`PartialEq`] implementation.
+///   A custom implementation may be provided using `#[reflect(PartialEq(my_partial_eq_func))]` where
+///   `my_partial_eq_func` is the path to a function matching the signature:
+///   `(&self, value: &dyn #bevy_reflect_path::Reflect) -> bool`.
+/// * `#[reflect(Hash)]` will force the implementation of `Reflect::reflect_hash` to rely on
+///   the type's [`Hash`] implementation.
+///   A custom implementation may be provided using `#[reflect(Hash(my_hash_func))]` where
+///   `my_hash_func` is the path to a function matching the signature: `(&self) -> u64`.
+/// * `#[reflect(Default)]` will register the `ReflectDefault` type data as normal.
+///   However, it will also affect how certain other operations are performed in order
+///   to improve performance and/or robustness.
+///   An example of where this is used is in the [`FromReflect`] derive macro,
+///   where adding this attribute will cause the `FromReflect` implementation to create
+///   a base value using its [`Default`] implementation avoiding issues with ignored fields.
+///
+/// ## `#[reflect_value]`
+///
+/// The `#[reflect_value]` attribute (which may also take the form `#[reflect_value(Ident)]`),
+/// denotes that the item should implement `Reflect` as though it were a base value type.
+/// This means that it will forgo implementing `Struct`, `TupleStruct`, or `Enum`.
+///
+/// Furthermore, it requires that the type implements [`Clone`].
+/// If planning to serialize this type using the reflection serializers,
+/// then the `Serialize` and `Deserialize` traits will need to be implemented and registered as well.
+///
+/// # Field Attributes
+///
+/// Along with the container attributes, this macro comes with some attributes that may be applied
+/// to the contained fields themselves.
+///
+/// ## `#[reflect(ignore)]`
+///
+/// This attribute simply marks a field to be ignored by the reflection API.
+///
+/// This allows fields to completely opt-out of reflection,
+/// which may be useful for maintaining invariants, keeping certain data private,
+/// or allowing the use of types that do not implement `Reflect` within the container.
+///
+/// ## `#[reflect(skip_serializing)]`
+///
+/// This works similar to `#[reflect(ignore)]`, but rather than opting out of _all_ of reflection,
+/// it simply opts the field out of both serialization and deserialization.
+/// This can be useful when a field should be accessible via reflection, but may not make
+/// sense in a serialized form, such as computed data.
+///
+/// What this does is register the `SerializationData` type within the `GetTypeRegistration` implementation,
+/// which will be used by the reflection serializers to determine whether or not the field is serializable.
+///
+/// [`reflect_trait`]: macro@reflect_trait
+#[proc_macro_derive(Reflect, attributes(reflect, reflect_value))]
```

Comment:
I like the detail, but this is too much. I think this needs to be condensed IMHO.